python/Ganga/Core/GangaRepository/SubJobXMLList.py

```diff
@@ -352,17 +351,28 @@ def _getItem(self, index):
                 if index in self._cachedJobs:
                     return self._cachedJobs[index]
 
+                has_loaded_backup = False
+
                 # Now try to load the subjob
                 if len(self) < index:
                     raise GangaException("Subjob: %s does NOT exist" % str(index))
                 subjob_data = self.__get_dataFile(str(index))
                 try:
                     sj_file = self._loadSubJobFromDisk(subjob_data)
-                except IOError as x:
-                    if x.errno == errno.ENOENT:
-                        raise IOError("Subobject %s not found: %s" % (index, x))
-                    else:
-                        raise RepositoryError(self,"IOError on loading subobject %s: %s" % (index, x))
+                except Exception as x:
+                    logger.warning("Error loading XML file: %s" % str(x))
+                    try:
+                        logger.debug("Loading subjob #%s from disk, recent changes may be lost" % index)
+                        subjob_data = self.__get_dataFile(str(index), True)
+                        sj_file = self._loadSubJobFromDisk(subjob_data)
+                        has_loaded_backup = True
+                    except Exception as err:
```

Comment:
Yes but at the cost of accepting an unknown number of different types or Exceptions. We need to sort out the GangaRepositoryXML's handling of exceptions first and make sure that the exceptions thrown from there are understood. Once I know this should only return 2 or 3 exceptions I will re-visit this.